init.lua

```diff
@@ -30,55 +30,42 @@ letters = {
 letter_cutter = {}
 letter_cutter.known_nodes = {}
 
-function letters.register_letters(modname, subname, from_node, description, tiles)
-	for _, row in ipairs(letters) do
-		local namel = subname.. "_letter_" ..row[1]
-		local nameu = subname.. "_letter_" ..row[2]
-		local descl = description.. " " ..row[3]
-		local descu = description.. " " ..row[4]
-		local tilesl = tiles.. "^letters_" ..row[1].. "_overlay.png^[makealpha:255,126,126"
-		local tilesu = tiles.. "^letters_" ..row[2].. "_overlay.png^[makealpha:255,126,126"
-		local groups = {not_in_creative_inventory=1, not_in_craft_guide=1, oddly_breakable_by_hand=1, attached_node=1}
-		minetest.register_node(":" ..modname..":"..namel, {
-			description = descl,
-			drawtype = "signlike",
-			tiles = {tilesl},
-			inventory_image = tilesl,
-			wield_image = tilesl,
-			paramtype = "light",
-			paramtype2 = "wallmounted",
-			sunlight_propagates = true,
-			is_ground_content = false,
-			walkable = false,
-			selection_box = {
-				type = "wallmounted",
-				--wall_top = <default>
-				--wall_bottom = <default>
-				--wall_side = <default>
-			},
-			groups = groups,
-			legacy_wallmounted = false,
-		})
-		minetest.register_node(":" ..modname..":"..nameu, {
-			description = descu,
-			drawtype = "signlike",
-			tiles = {tilesu},
-			inventory_image = tilesu,
-			wield_image = tilesu,
-			paramtype = "light",
-			paramtype2 = "wallmounted",
-			sunlight_propagates = true,
-			is_ground_content = false,
-			walkable = false,
-			selection_box = {
-				type = "wallmounted",
-				--wall_top = <default>
-				--wall_bottom = <default>
-				--wall_side = <default>
-			},
-			groups = groups,
-			legacy_wallmounted = false,
-		})
+function letters.register_letters(modname, subname, from_node, description, tiles, def)
+	--default node
+	def.drawtype = "signlike"
```

Comment:
This will modify the table which is submitted to the function. Also consider falling back to an empty table by default to make this parameter optional:
```Lua
def = def and table.copy(def) or {} -- drop table reference
```

Reply:
If def is not call this is one error.

One solution:

'''lua
function letters.register_letters(modname, subname, from_node, description, tiles)
	local def = {}
	function letters.register_letters_parametre(modname, subname, from_node, description, tiles, def)
end

function letters.register_letters_parametre(modname, subname, from_node, description, tiles, def)
...

Reply:
It would be an error in other programming languages when you don't pass the right amount of parameters to a function. Lua is different.
When you call `letters.register_letters("foo", "bar")` alone, then all parameters starting from `from_node` will be `nil`.

Reply:
Ok, thx.